front/components/assistant/ToolsPicker.tsx

```diff
@@ -163,32 +163,48 @@ export function ToolsPicker({
     };
   }, [serverViews, searchText, selectedMCPServerViewIds]);
 
-  const { availableMCPServers } = useAvailableMCPServers({
-    owner,
-  });
+  const { availableMCPServers, isAvailableMCPServersLoading } =
+    useAvailableMCPServers({
+      owner,
+      disabled: !shouldKeepHooksActive,
+    });
+
+  const isDataReady = !isServerViewsLoading && !isAvailableMCPServersLoading;
 
-  // We compare by name, not sId, because names are shared between multiple instances of the same MCP server (sIds are not).
-  // We filter by manual availability to show only servers that need install step.
-  const uninstalledServers = useMemo(() => {
-    if (!availableMCPServers || !serverViews) {
+  // - We compare by name, not sId, because names are shared between multiple instances of the same MCP server (sIds are not).
+  // - We filter by manual availability to show only servers that need install step, and by search text if present.
+  // - We don't compute uninstalled servers until BOTH data sources have loaded to prevent flicker.
+  const filteredUninstalledServers = useMemo(() => {
+    if (!hasFeature("jit_tool_setup") || !isAdmin || !isDataReady) {
       return [];
     }
+
     const installedServerNames = new Set(serverViews.map((v) => v.server.name));
-    return availableMCPServers.filter(
+    const uninstalled = availableMCPServers.filter(
```

Comment:
if "shouldKeepHooksActive" is false (name is not very clear imo ?) , serverViews and availableMCPServers  will be empty , so everthing will be empty, why not returning [] directly  by adding a condition on L178 ?